validation/src/shared_table/mod.rs

```diff
@@ -519,7 +523,17 @@ impl SharedTable {
 
 	/// Get all witnessed misbehavior.
 	pub fn get_misbehavior(&self) -> HashMap<SessionKey, table::Misbehavior> {
-		self.inner.lock().table.get_misbehavior().clone()
+		self
+			.inner
+			.lock()
+			.table
+			.get_misbehavior()
+			.iter()
+			.map(|(i, m)| {
+				let key = self.context.index_to_id(*i).expect("TODO: FIXME");
```

Comment:
again, would be simplified by having `Misbehavior` also use the index (if it's canonical, the runtime can easily do the same lookup when evaluating reports)

Reply:
Just to be clear: return value here can be `HashMap<ValidatorIndex, table::Misbehavior>`, right? Otherwise `ValidatorIndex` still needs to be converted to `SessionKey`, though `Misbehavior` can be returned as it is.